### PR TITLE
allow CartesianIndex in `isassigned`

### LINKED ISSUE #43392: isassigned not defined for CartesianIndex arguments
```julia
julia> M = rand(1:100, 4, 5)
4×5 Matrix{Int64}:
 88  94  82   2  78
 94  64  65  85  50
 35  99   9  84  96
 94  14  42  49  50

julia> isassigned(M, CartesianIndex(3, 2))
ERROR: MethodError: no method matching isassigned(::Matrix{Int64}, ::CartesianIndex{2})
Closest candidates are:
  isassigned(::Array, ::Int64...) at array.jl:201
  isassigned(::AbstractArray, ::Integer...) at abstractarray.jl:511
Stacktrace:
 [1] top-level scope
   @ REPL[45]:1

julia> isassigned(M, 7)
true

julia> versioninfo()
Julia Version 1.6.2
```

### PR DESCRIPTION
**Pull request by Moelf**
close #43392

### CODE REVIEW

**Comment by N5N3** [comment]
We'd better define these methods in `multidimensional.jl`.
BTW, why limit it to `::Array`? We also has a `try` based fallback for `AbstractArray`.

**Comment by Moelf** [comment]
they can already do it via:
```julia
isassigned(A, CartesianIndex(1, CartesianIndex(1,1)))
```

File: base/multidimensional.jl [hunk 1]
```diff
@@ -1534,6 +1534,9 @@ end
     end
 end
 
+isassigned(a::Array, i::CartesianIndex) = isassigned(a, to_indices(a, (), (i, ))...)
```
Comment (JeffBezanson) [inline base/multidimensional.jl:1537]:
```suggestion
isassigned(a::Array, i::CartesianIndex) = isassigned(a, Tuple(i)...)
```

**Review by JeffBezanson — APPROVED** [review]

**Comment by Moelf** [comment]
bump for 1.8 train

File: base/multidimensional.jl [hunk 2]
```diff
@@ -1534,6 +1534,9 @@ end
     end
 end
 
+isassigned(a::Array, i::CartesianIndex) = isassigned(a, Tuple(i)...)
+isassigned(a::Array, i::Union{Int, CartesianIndex}...) = isassigned(a, CartesianIndex(i))
```
Comment (N5N3) [inline base/multidimensional.jl:1538]: > BTW, why limit it to ::Array? We also has a try based fallback for AbstractArray.
```suggestion
isassigned(a::AbstractArray, i::CartesianIndex) = isassigned(a, Tuple(i)...)
isassigned(a::AbstractArray, i::Union{Int, CartesianIndex}...) = isassigned(a, CartesianIndex(i))
```

File: base/multidimensional.jl [hunk 3]
```diff
@@ -1534,6 +1534,9 @@ end
     end
 end
 
+isassigned(a::AbstractArray, i::CartesianIndex) = isassigned(a, Tuple(i)...)
+isassigned(a::AbstractArray, i::Union{Int, CartesianIndex}...) = isassigned(a, CartesianIndex(i))
```
Comment (N5N3) [inline base/multidimensional.jl:1538]:
```suggestion
isassigned(a::AbstractArray, i::Union{Integer, CartesianIndex}...) = isassigned(a, CartesianIndex(i))
```
This fix the stack overflow locally.

Comment (N5N3) [reply base/multidimensional.jl]: This stack overflow should be predictable. As the general fallback is defined as:
```julia
function isassigned(a::AbstractArray, i::Integer...)
```
Thus we encounter a circular subtype chain.
Have no idea on the test though.

Comment (Moelf) [reply base/multidimensional.jl]: you said "locally", just put that into tests?

Comment (N5N3) [reply base/multidimensional.jl]: Any `AbstractArray` using try based fallback should be testable. e.g. `isassigned(view([1],1:1),1)`.
But this should be tested implicitly via `show`. (unassigned value is printed as #undef in REPL)

Comment (Moelf) [reply base/multidimensional.jl]: alright, no tests then, testing for not stack overflowing is weird anyway

**Comment by Moelf** [comment]
genten bump

**Pull request merged**